func-names with ES6 classes

ESLint seems to misunderstand ES6 classes definitions.

test.js :
```js
class A {
  constructor() {}
  foo() {}
}
```

.eslintrc
```yaml
ecmaFeatures:
  classes: true
rules:
  func-names: 2
```

Errors :
```
test.js
  2:13  error  Missing function expression name  func-names
  3:5   error  Missing function expression name  func-names
```
I can make probably make a patch but I don't know if the problem is in the parser or in ESLint.

--- a/lib/rules/func-names.js
+++ b/lib/rules/func-names.js
@@ -15,15 +15,18 @@ module.exports = function(context) {
 
     /**
      * Determines whether the current FunctionExpression node is a get, set, or
-     * shorthand method in an object literal.
+     * shorthand method in an object literal or a class.
      * @returns {boolean} True if the node is a get, set, or shorthand method.
      */
-    function isObjectMethod() {
+    function isObjectOrClassMethod() {
         var parent = context.getAncestors().pop();
-        return (parent.type === "Property" && (
-            parent.method ||
-            parent.kind === "get" ||
-            parent.kind === "set"
+
+        return (parent.type === "MethodDefinition" || (
+            parent.type === "Property" && (
+                parent.method ||
+                parent.kind === "get" ||
+                parent.kind === "set"
+            )
         ));
     }
 
@@ -32,7 +35,7 @@ module.exports = function(context) {
 
             var name = node.id && node.id.name;
 
-            if (!name && !isObjectMethod()) {
+            if (!name && !isObjectOrClassMethod()) {
                 context.report(node, "Missing function expression name.");
             }
         }

--- a/tests/lib/rules/func-names.js
+++ b/tests/lib/rules/func-names.js
@@ -30,6 +30,10 @@ eslintTester.addRuleTest("lib/rules/func-names", {
         {
             code: "({ foo() { return 1; } });",
             ecmaFeatures: { objectLiteralShorthandMethods: true }
+        },
+        {
+            code: "class A { constructor(){} foo(){} get bar(){} set baz(value){} static qux(){}}",
+            ecmaFeatures: { classes: true }
         }
     ],
     invalid: [

